### PR TITLE
Add k8s infra monitorig demo example

### PR DESCRIPTION
**Pull request by ChrsMark**
This PR adds a new example for the `opentelemetry-demo` chart.
This example aims to provide a way to enable Kubernetes cluster monitoring by setting the proper collector's components.

Note that only node level metrics can be enabled because `clusterMetrics` requires `deployment` mode instead of `daemonset`. However, it would be nice if we could combine `clusterMetrics` as well so any ideas more than welcome. I guess this would require to edit the chart a bit so as to add an additional collector running as Deployment?

### CODE REVIEW

File: charts/opentelemetry-demo/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 type: application
 name: opentelemetry-demo
-version: 0.26.0
+version: 0.27.0
```
Comment (JaredTan95) [inline charts/opentelemetry-demo/Chart.yaml:4]: This is not a functional PR, I don't think we need to update the minor version, how about `0.26.1`?

**Review by JaredTan95 — APPROVED** [review]

**Comment by ChrsMark** [comment]
Fair point @povilasv, but if we think that the demo can also run on k8s I find value into combining traces of the services with the infrastructure metrics from the cluster the services are running on. In this we have a full stack Observability demonstration for Otel.

**Review by Allex1 — APPROVED** [review]
This is pretty awesome @ChrsMark .
Thanks

**Review by TylerHelmuth — COMMENTED** [review]
Since our CI doesn't test using examples, can you confirm @ChrsMark that the new example produces a working instance of the demo?

**Comment by ChrsMark** [comment]
Sure. Here is a complete working example that I used:

```yml
opentelemetry-collector:
  mode: "daemonset"
  presets:
    kubernetesAttributes:
      enabled: true
    kubeletMetrics:
      enabled: true
    hostMetrics:
      enabled: true
    logsCollection:
      enabled: true
      includeCollectorLogs: false
      storeCheckpoints: true

  extraEnvs:
    - name: ELASTIC_APM_ENDPOINT
      valueFrom:
        secretKeyRef:
          name: elastic-secret
          key: elastic_apm_endpoint
    - name: ELASTIC_APM_SECRET_TOKEN
      valueFrom:
        secretKeyRef:
          name: elastic-secret
          key: elastic_apm_secret_token
  config:
    exporters:
      otlp/elastic:
        endpoint: ${ELASTIC_APM_ENDPOINT}
        compression: none
        headers:
          Authorization: Bearer ${ELASTIC_APM_SECRET_TOKEN}
      otlp:
        protocols:
          grpc:
            endpoint: ${MY_POD_IP}:4317
          http:
            cors:
              allowed_origins:
              - http://*
              - https://*
            endpoint: ${MY_POD_IP}:4318
    service:
      pipelines:
        traces:
          receivers: [otlp]
          processors: [batch]
          exporters: [otlp/elastic]
        metrics:
          receivers: [otlp]
          processors: [batch]
          exporters: [otlp/elastic]
        logs:
          receivers: [otlp]
          processors: [batch]
          exporters: [otlp/elastic]
grafana:
  enabled: false

jaeger:
  enabled: false

prometheus:
  enabled: false
```

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**